### ISSUE TITLE
IndexError: index 1 is out of bounds for axis 0 with size 1

### ISSUE THREAD

**Issue by ktsitsi** [cgohlke/tifffile#253]
I am getting the following error traceback, while trying to read a `.tiff` from [Camelyon16](https://registry.opendata.aws/camelyon/). The image I am testing is the  `s3://camelyon-dataset/CAMELYON16/images/normal001.tif`

I am running this on an EC2 machine with sufficient memory `137Gib` available. I notice that the memory during mem fetching max out at around `60GB` and then this error pops up. The shape of the level[0] is `(220452, 97469, 3)` with a dtype `uint8` ~= 60Gb. I am using tifffile version `'2024.4.24'`.

```bash
In [1]: import tifffile
In [2]: tiff_f=tifffile.TiffFile('./normal_001.tif')
In [3]: series=tiff_f.series[0]
In [4]: series.levels[0].asarray()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-c7f1fb5d8b1d> in ?()
----> 1 series.levels[0].asarray()

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(self, level, **kwargs)
  12295         if self.parent is None:
  12296             raise ValueError('no parent')
  12297         if level is not None:
  12298             return self.levels[level].asarray(**kwargs)
> 12299         result = self.parent.asarray(series=self, **kwargs)
  12300         if self.transform is not None:
  12301             result = self.transform(result)
  12302         return result

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(self, key, series, level, squeeze, out, maxworkers, buffersize)
   4523         elif len(pages) == 1:
   4524             page0 = pages[0]
   4525             if page0 is None:
   4526                 raise ValueError('page is None')
-> 4527             result = page0.asarray(
   4528                 out=out, maxworkers=maxworkers, buffersize=buffersize
   4529             )
   4530         else:

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(self, out, squeeze, lock, maxworkers, buffersize)
   9340                     ]
   9341                 # except IndexError:
   9342                 #     pass  # corrupted file, for example, with too many strips
   9343 
-> 9344             for _ in self.segments(
   9345                 func=func,
   9346                 lock=lock,
   9347                 maxworkers=maxworkers,

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(self, lock, maxworkers, func, sort, buffersize, _fullsize)
   9150                     sort=sort,
   9151                     buffersize=buffersize,
   9152                     flat=False,
   9153                 ):
-> 9154                     yield from executor.map(decode, segments)

/usr/lib/python3.10/concurrent/futures/_base.py in ?()
    621                         yield _result_or_cancel(fs.pop())
    622                     else:
    623                         yield _result_or_cancel(fs.pop(), end_time - time.monotonic())
    624             finally:
--> 625                 for future in fs:
    626                     future.cancel()

/usr/lib/python3.10/concurrent/futures/_base.py in ?(***failed resolving arguments***)
    320         finally:
    321             fut.cancel()
    322     finally:
    323         # Break a reference cycle with the exception in self._exception
--> 324         del fut

/usr/lib/python3.10/concurrent/futures/_base.py in ?(self, timeout)
    459                 else:
    460                     raise TimeoutError()
    461         finally:
    462             # Break a reference cycle with the exception in self._exception
--> 463             self = None

/usr/lib/python3.10/concurrent/futures/_base.py in ?(self)
    402             try:
    403                 raise self._exception
    404             finally:
    405                 # Break a reference cycle with the exception in self._exception
--> 406                 self = None
    407         else:
    408             return self._result

/usr/lib/python3.10/concurrent/futures/thread.py in ?(self)
     58             result = self.fn(*self.args, **self.kwargs)
     59         except BaseException as exc:
     60             self.future.set_exception(exc)
     61             # Break a reference cycle with the exception 'exc'
---> 62             self = None
     63         else:
     64             self.future.set_result(result)

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(args, decodeargs, decode)
   9126             def decode(args, decodeargs=decodeargs, decode=keyframe.decode):
-> 9127                 return func(decode(*args, **decodeargs))  # type: ignore

~/.local/lib/python3.10/site-packages/tifffile/tifffile.py in ?(decoderesult, keyframe, out)
   9330                     out[
   9331                         s, d : d + shape[0], h : h + shape[1], w : w + shape[2]
   9332                     ] = keyframe.nodata
   9333                 else:
-> 9334                     out[
   9335                         s, d : d + shape[0], h : h + shape[1], w : w + shape[2]
   9336                     ] = segment[
   9337                         : keyframe.imagedepth - d,

IndexError: index 1 is out of bounds for axis 0 with size 1

```

**Comment by cgohlke** [comment]
Thanks for reporting. I can reproduce the issue. It is closely related to #249. In that case the Philips formatted TIFF had an entire row of tiles missing, while this file has an extra column of tiles (compared to what is expected from the Philips XML metadata). Probably the interpretation of the Philips XML metadata needs to be revised...

**Comment by cgohlke** [comment]
I'm not sure yet what's the best fix. Probably giving up to determine the padding of the base level...

In the meantime, if you are only interested in the base level, try to disable the Philips handling,`TiffFile('normal_001.tif', is_philips=False)`

**Comment by cgohlke** [comment]
Fixed in tifffile-2024.5.3.
